integration/experiment/system_characterization/gen_cpu_activity_constconfig_recommendation.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2015 - 2025 Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+'''
+Finds the energy efficient frequency for a provided frequency sweep
+and characterizes the system memory bandwidth usage for the CPU Activity Agent
+as a ConstConfigIO configuration file.
+'''
+
+import argparse
+import json
+import os
+import sys
+
+import pandas
+import numpy as np
+
+import geopmpy.io
+
+from integration.experiment import util
+from integration.experiment import common_args
+from integration.experiment import machine
+
+def extract_columns(df, region_list = None):
+    """
+    Extract the columns of interest from the full report collection
+    dataframe.
+    """
+
+    if region_list:
+        #Explicitly a copy to deal with setting on copy errors
+        df_filtered = df.loc[df['region'].isin(region_list)].copy()
+    else:
+        #Explicitly a copy to deal with setting on copy errors
+        df_filtered = df.copy()
+
+    # These cases are intended to handle older runs where uncore-frequency is not
+    # part of the report or the QM_CTR_SCALED_RATE has not been collected at the
+    # board level, and should be removed at a later date
+
+    if 'MSR::QM_CTR_SCALED_RATE' not in df_filtered.columns:
```

Comment:
This might be a good place to confirm that you have all the columns needed and print out a clear error message if they aren't present.